tests/rgen/Microsoft.Macios.Generator.Tests/DataModel/BindingTests.cs

```diff
@@ -259,6 +260,361 @@ public void SkipMethodDeclaration (ApplePlatform platform, string inputText, boo
+			yield return [mixedSupport, ApplePlatform.iOS, false, 2, 2, 2];
+
+			// No constructors supported in platform
+			var noConstructorsSupported = @"
+using System;
+using System.Runtime.Versioning;
+using Foundation;
+using ObjCBindings;
+using ObjCRuntime;
+
+namespace NS {
+	[BindingType<Class>]
+	public partial class MyClass {
+		[UnsupportedOSPlatform (""tvos"")]
+		[Export<Constructor> (""init"")]
+		public MyClass ();
+
+		[UnsupportedOSPlatform (""tvos"")]
+		[Export<Constructor> (""init2"")]
+		public MyClass (string value);
+
+		[Export<Method> (""method1"")]
+		public virtual partial void Method1 ();
+
+		[Export<Property> (""property1"")]
+		public virtual partial string Property1 { get; set; }
+	}
+}
+";
+			yield return [noConstructorsSupported, ApplePlatform.TVOS, false, 0, 1, 1];
+
+			// No methods supported in platform
+			var noMethodsSupported = @"
+using System;
+using System.Runtime.Versioning;
+using Foundation;
+using ObjCBindings;
+using ObjCRuntime;
+
+namespace NS {
+	[BindingType<Class>]
+	public partial class MyClass {
+		[Export<Constructor> (""init"")]
+		public MyClass ();
+
+		[UnsupportedOSPlatform (""tvos"")]
+		[Export<Method> (""method1"")]
+		public virtual partial void Method1 ();
+
+		[UnsupportedOSPlatform (""tvos"")]
+		[Export<Method> (""method2"")]
+		public virtual partial void Method2 ();
+
+		[Export<Property> (""property1"")]
+		public virtual partial string Property1 { get; set; }
+	}
+}
+";
+			yield return [noMethodsSupported, ApplePlatform.TVOS, false, 1, 0, 1];
+
+			// No properties supported in platform
+			var noPropertiesSupported = @"
+using System;
+using System.Runtime.Versioning;
+using Foundation;
+using ObjCBindings;
+using ObjCRuntime;
+
+namespace NS {
+	[BindingType<Class>]
+	public partial class MyClass {
+		[Export<Constructor> (""init"")]
+		public MyClass ();
+
+		[Export<Method> (""method1"")]
+		public virtual partial void Method1 ();
+
+		[UnsupportedOSPlatform (""tvos"")]
+		[Export<Property> (""property1"")]
+		public virtual partial string Property1 { get; set; }
+
+		[UnsupportedOSPlatform (""tvos"")]
+		[Export<Property> (""property2"")]
+		public virtual partial string Property2 { get; set; }
+	}
+}
+";
+			yield return [noPropertiesSupported, ApplePlatform.TVOS, false, 1, 1, 0];
+
+			// Platform-specific availability with iOS version constraints
+			var versionConstraints = @"
+using System;
+using System.Runtime.Versioning;
+using Foundation;
+using ObjCBindings;
+using ObjCRuntime;
+
+namespace NS {
+	[BindingType<Class>]
+	public partial class MyClass {
+		[Export<Constructor> (""init"")]
+		public MyClass ();
+
+		[SupportedOSPlatform (""ios15.0"")]
+		[Export<Method> (""newMethod"")]
+		public virtual partial void NewMethod ();
+
+		[SupportedOSPlatform (""ios14.0"")]
+		[Export<Property> (""newProperty"")]
+		public virtual partial string NewProperty { get; set; }
+	}
+}
+";
+			yield return [versionConstraints, ApplePlatform.iOS, false, 1, 1, 1];
```

Comment:
I'm not sure how your code handles this, but if there are SupportedOSPlatform attributes, then any platform not mentioned in a SupportedOSPlatform attribute are not supported. So for instance for this test (versionConstraints), while the class and the constructors are available on all platforms, the method and the property are only available on iOS.

So I think something like this would add test coverage for this scenario:

```suggestion
			yield return [versionConstraints, ApplePlatform.iOS, false, 1, 1, 1];
			yield return [versionConstraints, ApplePlatform.TVOS, false, 1, 0, 0];
```